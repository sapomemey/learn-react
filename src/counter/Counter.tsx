import React, { useState } from 'react';
import './counter.css';

const Counter = ({ initialValue }: any) => {
  const [num, setNum] = useState<number>(initialValue);

  const hanlderAdd = () => setNum(currentValue => currentValue + 1);
  const handlerLess = () => setNum(currentValue => currentValue - 1);
  const handlerReset = () => setNum(initialValue);

  return <>
    <div>{num}</div>
    <button onClick={hanlderAdd}>add</button>
    <button onClick={handlerReset}>reset</button>
    <button onClick={handlerLess}>less</button>
  </>;
};

export default Counter;
