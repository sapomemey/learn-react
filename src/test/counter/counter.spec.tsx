import { shallow } from 'enzyme';
import Counter from '../../counter/Counter';

describe('@bcp testing counter', () => {
  it('should render', () => {
    const wrapper = shallow(<Counter initialValue={5}></Counter>);

    expect(wrapper).toMatchSnapshot();
  });

  it('should sum one when is clicked in button add', () => {
    const wrapper = shallow(<Counter initialValue={2}></Counter>);
    wrapper.find('button').at(0).simulate('click');
    const num = wrapper.find('div').text().trim();

    expect(num).toBe('3');
  });


  it('should less one when is clicked in button less', () => {
    const wrapper = shallow(<Counter initialValue={2}></Counter>);
    wrapper.find('button').at(2).simulate('click');
    const num = wrapper.find('div').text().trim();

    expect(num).toBe('1');
  });

  it('should reset to default value when is clicked in button reset', () => {
    const wrapper = shallow(<Counter initialValue={6}></Counter>);
    wrapper.find('button').at(0).simulate('click');
    wrapper.find('button').at(0).simulate('click');
    wrapper.find('button').at(0).simulate('click');
    wrapper.find('button').at(1).simulate('click');

    const num = wrapper.find('div').text().trim();

    expect(num).toBe('6')
  });

});
