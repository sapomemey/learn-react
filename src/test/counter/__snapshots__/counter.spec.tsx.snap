// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@bcp testing counter should render 1`] = `
<Fragment>
  <div>
    5
  </div>
  <button
    onClick={[Function]}
  >
    add
  </button>
  <button
    onClick={[Function]}
  >
    reset
  </button>
  <button
    onClick={[Function]}
  >
    less
  </button>
</Fragment>
`;
